api-get-object-file.go

```diff
@@ -21,11 +21,18 @@ import (
 	"os"
 	"path/filepath"
 
+	"context"
+
 	"github.com/minio/minio-go/pkg/s3utils"
 )
 
 // FGetObject - download contents of an object to a local file.
 func (c Client) FGetObject(bucketName, objectName, filePath string) error {
+	return c.fGetObjectWithContext(context.Background(), bucketName, objectName, filePath)
```

Comment:
I am not sure how it is useful to provide two APIs which has the same functionality, for example here:
- FGetObject
and
- FGetObjectWithContext

I think we need to decide if they should have different behaviors or merge them into one function

Reply:
It was decided to have an extra variant of Put/GetObject when context is an additional argument, rather than force user to pass nil each time if they do not need request cancellation. We could however revisit this in the next meeting and reach a consensus.

Reply:
> rather than force user to pass nil each time if they do not need request cancellation.

emm, in that case, we need to pass nil here (if it works) and not context.Background()

Reply:
as per https://golang.org/pkg/context/
Do not pass a nil Context, even if a function permits it. Pass context.TODO if you are unsure about which Context to use. 

Since we are wrapping the functions, it makes sense to use context.Background(). Agree?

Reply:
> Since we are wrapping the functions, it makes sense to use context.Background(). Agree?

sorry, yeah I see now how it can be useful to have two APIs